### PR TITLE
Start delete secret

### PR DESCRIPTION
**Pull request by gearama**
Start Delete  and Start recover secret operations, also moved a couple of the option structs to a separate faile form the client to make the file  less crowded
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [ ] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [ ] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [ ] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_operations.hpp [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            m_status = Azure::Core::OperationStatus::Running;
+            break;
+          }
+          default:
+            throw Azure::Core::RequestFailedException(rawResponse);
+        }
+
+        if (m_status == Azure::Core::OperationStatus::Succeeded)
+        {
+          m_value = _detail::KeyVaultSecretSerializer::KeyVaultSecretDeserialize(
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_operations.hpp:86]: Move this implementation to cpp to call the Deserializer

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_options.hpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+   */
+  struct UpdateSecretPropertiesOptions final
+  {
+    /**
+     * @brief Specify the secret version to update.
+     *
+     */
+    std::string Version;
+  };
+}}}} // namespace Azure::Security::KeyVault::Secrets
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_options.hpp:90]:
```suggestion
}}}} // namespace Azure::Security::KeyVault::Secrets

```

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp [hunk 3]
```diff
@@ -22,85 +23,13 @@
 namespace Azure { namespace Security { namespace KeyVault { namespace _detail {
   class KeyVaultProtocolClient;
 }}}} // namespace Azure::Security::KeyVault::_detail
-
 namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
+  class KeyVaultRestoreDeletedSecretOperation;
+  class KeyVaultDeleteSecretOperation;
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:28]: Why?  These types should be public.  just include the header directly.

Look at the latest main for KeyVault keys, Operations is inside of the Models.hpp header and is just included in the client

**Review by vhvb1989 — COMMENTED** [review]
Update the client to include the Operations from header to avoid the uncompleted type class.

Some other comments over there for you to take a look

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_options.hpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Keyvault Secret actions options
+ */
+#pragma once
+#include "azure/keyvault/secrets/dll_import_export.hpp"
+#include <azure/core.hpp>
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_options.hpp:10]: really?  All core? Why?

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp [hunk 5]
```diff
@@ -289,6 +213,34 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
...
+        Azure::Core::Context const& context = Azure::Core::Context()) const;
+
+    /**
+     * @brief Recover the deleted secret to the latest version.
+     *
+     * @remark Recovers the deleted secret in the specified vault.
+     * This operation can only be performed on a soft-delete enabled vault.
+     * This operation requires the secrets/recover permission.
+     *
+     * @param name The name of the secret<span class="x x-first x-last">.</span>
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:238]: Are you adding this html tag `span class=...`  for the dot?  I have seen it before.   I wonder what's this is about?

**Review by vhvb1989 — COMMENTED** [review]
few comments.
And why are we not adding unit tests this time?

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_options.hpp (hunk 4 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_options.hpp:10]: fixed

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp (hunk 5 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp]: it came from a comment in a PR a long time ago, so it just got perpetuated through copy paste i don't think it's of any use. I removed it for all occurences.

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**